src/components/forms/ClusteredDiskSizeSelector.tsx

```diff
@@ -0,0 +1,231 @@
+      const newDefaultSpecific = rawValues.some(
+        (item) => item !== rawValues[0],
+      );
+      setIsSpecific(newDefaultSpecific);
+    }
+  }, [isSpecific, sizeValues]);
+
+  const setValueForAllMembers = (value: string) => {
+    const update: ClusterSpecificValues = {};
+    memberNames.forEach((member) => (update[member] = value));
+    setMemoryLimit(update);
+  };
+
+  const setValueForMember = (value: string, member: string) => {
+    const update = {
+      ...sizeValues,
+      [member]: value,
+    };
+    setMemoryLimit(update);
+  };
+
+  return (
+    <div className="u-sv3">
+      <Label forId="poolSizePerClusterMember">Size</Label>
+      {canToggleSpecific && !isReadOnly && (
+        <CheckboxInput
+          id={`${id}-same-for-all-toggle`}
+          label="Same for all cluster members"
+          checked={!isSpecific}
+          onChange={() => {
+            setIsSpecific((val) => !val);
+          }}
+        />
+      )}
+      {isSpecific && (
+        <div className="cluster-specific-input">
+          {memberNames.map((item) => {
+            const activeValue = sizeValues?.[item];
+
+            return (
+              <Fragment key={item}>
+                <div className="cluster-specific-member">
+                  <ResourceLink
+                    type="cluster-member"
+                    value={item}
+                    to={clusterMemberLinkTarget(item)}
+                  />
+                </div>
+                <div className="cluster-specific-value">
+                  {isReadOnly ? (
+                    <>
+                      {activeValue}
+                      <FormEditButton toggleReadOnly={toggleReadOnly} />
+                    </>
+                  ) : (
+                    <div className="memory-limit-with-unit">
+                      <Input
+                        id={
+                          memberNames.indexOf(item) === 0 ? id : `${id}-${item}`
+                        }
+                        name="limits_disk"
+                        type="number"
+                        min="0"
+                        step="Any"
+                        placeholder="Enter value"
+                        onChange={(e) =>
+                          setValueForMember(
+                            e.target.value + limit(activeValue || "").unit,
+                            item,
+                          )
+                        }
+                        value={
+                          activeValue?.match(/^\d/)
+                            ? limit(activeValue || "").value
+                            : ""
+                        }
+                        className="u-no-margin--bottom"
+                      />
+                      <Select
+                        id={
+                          memberNames.indexOf(item) === 0
+                            ? "memUnitSelect"
+                            : `memUnitSelect-${item}`
```

Comment:
I think we should append the id prop from this component. We might use this component more than once per page and with the current code it would generate duplicate ids.